apps/api/src/controllers/posts.ts

```diff
@@ -1,155 +1,91 @@
-  }
-  if (category) {
-    post.category.name = category;
-  }
-
-  posts[postIndex] = post;
-
-  return { isInvalid: false, post };
-};
 
 // - `GET /posts` Return an array of all the posts with status code 200
-const getAllPosts = (req, res) => {
-  res.status(200).json(posts);
+const getAllPosts = async (req, res) => {
+  try {
+    const categories = await Post.find({});
+    res.status(200).json(categories);
+  } catch (error) {
+    const { message } = error;
+    res.status(500).json({ message });
+  }
 };
 
 // - `GET /posts/category/:category` Return an array of all the posts by category with status code 200
-const getPostByCategory = (req, res) => {
-  const { category } = req.params;
-  const postsByCategory = getCategory(category);
-
-  res.status(200).json(postsByCategory);
+const getPostByCategory = async (req, res) => {
+  try {
+    const postsByCategory = await Post.find({ category: req.params.category }).populate('category');
+    res.status(200).json(postsByCategory);
+  } catch (error) {
+    res.status(404).json({ message: 'No Post found by the selected category' });
+  }
 };
 // - `GET /posts/:id` Return a post by id with category object and each comment object in the array with status code 200
-const getPostById = (req, res) => {
+const getPostById = async (req, res) => {
   const { id } = req.params;
-  const postById = getPostId(id);
 
-  if (!postById) {
+  try {
+    const postById = await Post.findById(id).populate('category');
+    res.status(200).json(postById);
+  } catch (error) {
     return res.status(404).json({ message: POST_NO_FOUND });
   }
-  res.status(200).json(postById);
 };
 // - `POST /posts` Create a new post and return the created post with status code 201
-const createPost = (req, res) => {
-  if (!isPostValid(req.body)) {
-    return res.status(400).json({ message: INVALID_POST });
+const createPost = async (req, res) => {
+  try {
+    const newPost = await Post.create(req.body);
+    res.status(201).json(newPost);
+  } catch (error) {
+    return res.status(400).json({ message: error.message });
   }
-
-  const newPost = createNewPost(req.body);
-  posts.push(newPost);
-  res.status(201).json(newPost);
 };
 // - `POST /posts/:id/comments` Create a comment inside the post and return the comment with status code 201
-const addPostComment = (req, res) => {
-  const { id } = req.params;
-  const { content } = req.body;
-  const isContentValid = content?.trim()?.length > 0;
-  const postIndex = getIndexById(posts, id);
-  if (postIndex < 0) {
-    return res.status(404).json({ message: POST_NO_FOUND });
-  } else if (!isContentValid) {
-    return res.status(400).json({ message: INVALID_COMMENT });
+const addPostComment = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const { content, author } = req.body;
+    const isContentValid = content?.trim()?.length > 0;
+    if (!isContentValid) {
+      return res.status(400).json({ message: INVALID_COMMENT });
+    }
```

Comment:
We could avoid this manual validations and the generation of the comment id if we follow the approach of having a comment model